index.html

```diff
@@ -696,6 +751,57 @@
         <li>When the request has been completed, resolve <var>promise</var> with <code>true</code>.
         </li>
       </ol>
+      <p>
+        The serializer for a <a><code>PushSubscription</code></a> invokes the following steps:
+      </p>
+      <ol>
+        <li>Let <var>map</var> be an empty map.
+        </li>
+        <li>Add an entry to <var>map</var> whose key name is <code>endpoint</code> and whose value
+        is the result of converting the <code>endpoint</code> attribute of the
+        <a><code>PushSubscription</code></a> to a serialized value.
+        </li>
+        <li>Let <var>keys</var> be an empty map.
+        </li>
+        <li>For each identifier <var>i</var> corresponding to keys in internal slots on the
+        <a><code>PushSubscription</code></a>, ordered by the name of the key:
+          <ol>
+            <li>Add an entry to <var>keys</var> whose key name is the name of the key and whose
+            value is an <code>ArrayBuffer</code> containing the defined serialization of the public
+            key (for an asymmetric key pair) or the key (for a symmetric key).
+            </li>
+          </ol>
+        </li>
+        <li>Add an entry to <var>map</var> whose key name is <code>keys</code> and whose value is
+        <var>keys</var>.
+        </li>
+        <li>Return <var>map</var>.
+        </li>
+      </ol>
+      <section>
+        <h2>
+          <a>PushEncryptionKeyName</a> enumeration
+        </h2>
+        <p>
+          Encryption keys used for <a>push message</a> encryption are provided to a <a>webapp</a>
+          through the <code>getKey</code> method or the serializer of
+          <code><a>PushSubscription</a></code>. Each key is named using a value from the
+          <code><dfn>PushEncryptionKeyName</dfn></code> enumeration.
+        </p>
+        <dl title="enum PushEncryptionKeyName" class="idl">
+          <dt>
+            p256dh
+          </dt>
+        </dl>
+        <p>
+          The <code><dfn id="widl-PushEncryptionKeyName-p256dh">p256dh</dfn></code> value is used
+          to retrieve the P-256 ECDH Diffie-Hellman public key described in
+          [[!WEBPUSH-ENCRYPTION]].
+        </p>
+        <p>
+          Other encryption keys could be provided to applications.
```

Comment:
Do you explicitly want to allow for UA-specific extensions to the PushEncryptionKeyName values? (Why?)